src/cpp/session/modules/SessionHTMLPreview.cpp

```diff
@@ -1009,8 +1009,9 @@ SEXP rs_showPageViewer(SEXP urlSEXP, SEXP titleSEXP)
       if (boost::algorithm::starts_with(url, "http"))
       {
          // verify local url only
-         if (!boost::algorithm::starts_with(url, "http://localhost") &&
-             !boost::algorithm::starts_with(url, "http://127.0.0.1"))
+         boost::regex re("^http[s]?://(?:localhost|127\\.0\\.0\\.1):([0-9]+).*$");
```

Comment:
This regex requires that the URL have a port component -- is that intentional? (seems like a localhost webserver on the default :80 is possible)